internal/fs/buffer/inmemory_write_buffer.go

```diff
@@ -15,33 +15,117 @@
 package buffer
 
 import (
-	"bytes"
+	"fmt"
 )
 
+// InMemoryWriteBuffer implements WriteBuffer and is stored in memory.
 type InMemoryWriteBuffer struct {
-	// Holds the data to be written to GCS. At any time, Buffer holds the data for
-	// 2 GCS write calls.
-	buffer *bytes.Buffer
+	// Holds the incoming data from kernel write calls.
+	currentBuffer []byte
+	// [ currentBufferStartOffset, currentBufferEndOffset) is the range of file offset for which the
+	// data is currently stored in currentBuffer.
+	currentBufferStartOffset int64
```

Comment:
Instead of startOffset and endOffset, start and end would suffice. Or you could create an range struct and use that for all sorts of offsets.

```golang
type Range struct {
  start, end int64
}

type ChunkBuffer struct
{
  ...
  contentOffset Range
  ...
}
```

Reply:
range is a reserved keyword so used offset instead, thanks :)